packages/core/upload/server/services/upload.js

```diff
@@ -121,15 +122,29 @@ module.exports = ({ strapi }) => ({
     );
     currentFile.getStream = () => fs.createReadStream(file.path);
 
-    const { optimize, isOptimizableImage } = strapi.plugin('upload').service('image-manipulation');
+    const { optimize, isImage, isFaultyImage, isOptimizableImage } = strapi
+      .plugin('upload')
+      .service('image-manipulation');
 
-    if (!(await isOptimizableImage(currentFile))) {
-      return currentFile;
+    if (await isImage(currentFile)) {
+      if (await isFaultyImage(currentFile)) {
+        throw new ApplicationError('File is not a valid image');
+      }
+      if (!(await isOptimizableImage(currentFile))) {
+        return currentFile;
+      }
     }
-
     return optimize(currentFile);
```

Comment:
I think there is an error here. In the case the file isn't an image, it will be optimized whereas it should not.

Reply:
you are right. Fixed!
Tested it for some pdfs and images and it seems to work ok.